src/wifi.rs

```diff
@@ -1895,6 +1903,60 @@ impl TryFrom<&WpsCredentialsRef> for WpsCredentials {
     }
 }
 
+impl ApStaConnectedRef {
+    /// MAC address of the station connected to ESP32 soft-AP
+    pub fn mac(&self) -> [u8; 6] {
+        self.0.mac
+    }
+
+    /// the aid that ESP32 soft-AP gives to the station connected to
+    pub fn aid(&self) -> u8 {
+        self.0.aid
+    }
+
+    /// flag to identify mesh child
+    pub fn is_mesh_child(&self) -> bool {
+        self.0.is_mesh_child
+    }
+}
+
+impl fmt::Debug for ApStaConnectedRef {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("ApStaConnectedRef")
+            .field("mac", &self.mac())
+            .field("aid", &self.aid())
+            .field("is_mesh_child", &self.is_mesh_child())
+            .finish()
+    }
+}
+
+impl ApStaDisconnectedRef {
```

Comment:
does the C disconnect struct also support an extra field "reason" if we look [here](https://github.com/espressif/esp-idf/blob/4f3cd0deb9c79c8282da4938a29d265705a57564/components/esp_wifi/include/esp_wifi_types_generic.h#L857)? 

Since otherwise the structs would be pretty much identical and we could have unified them.

Reply:
Hmm, interesting. I was only looking into the official 4.3.7 documentation and it does not include this reason field, but newer versions do include it. It seems to have been added in 5.1.3

Reply:
Okay, I managed to add it via cfg attributes.
As the exact type of this reason is not specified either in the bindings nor in the documentation, I would assume it is a subset of [wifi_err_reason_t](https://github.com/espressif/esp-idf/blob/4f3cd0deb9c79c8282da4938a29d265705a57564/components/esp_wifi/include/esp_wifi_types_generic.h#L78-L140).
This is probably supposed to be mapped to an enum, but it seems to me that these types of structs and enums are maintained in https://github.com/esp-rs/embedded-svc and a From/TryFrom implementation for Newtype<Binding> is created in the esp-idf-svc project.

I should probably create another PR in the embedded-svc project to add the enum for `wifi_err_reason_t` right?